File: golang/uPIMulator/src/linker/kernel/instruction/cc/shift_nz_cc.go
package cc

import (
	"errors"
)

type ShiftNzCc struct {
	condition Condition
}

func (this *ShiftNzCc) Init(condition Condition) {
	conditions := map[Condition]bool{
		Z:     true,
		NZ:    true,
		XZ:    true,
		XNZ:   true,
		E:     true,
		O:     true,
		PL:    true,
		MI:    true,
		SZ:    true,
		SNZ:   true,
		SE:    true,
		SO:    true,
		SPL:   true,
		SMI:   true,
		SH32:  true,
		NSH32: true,
		TRUE:  true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *ShiftNzCc) Condition() Condition {
	return this.condition
}

File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/shift_nz_cc.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class ShiftNZCC(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(
            ShiftNZCC.conditions(), condition,
        )

    @staticmethod
    def conditions() -> Set[Condition]:
        return {
            Condition.Z,
            Condition.NZ,
            Condition.XZ,
            Condition.XNZ,
            Condition.E,
            Condition.O,
            Condition.PL,
            Condition.MI,
            Condition.SZ,
            Condition.SNZ,
            Condition.SE,
            Condition.SO,
            Condition.SPL,
            Condition.SMI,
            Condition.SH32,
            Condition.NSH32,
            Condition.TRUE,
        }
